src/Components/Components/src/Rendering/HtmlRenderer.cs

```diff
@@ -109,28 +109,27 @@ private int RenderFrames(List<string> result, ArrayRange<RenderTreeFrame> frames
         }
 
         private int RenderCore(
-            List<string> result,
+            HtmlRenderingContext context,
             ArrayRange<RenderTreeFrame> frames,
-            int position,
-            int length)
+            int position)
         {
             ref var frame = ref frames.Array[position];
             switch (frame.FrameType)
             {
                 case RenderTreeFrameType.Element:
-                    return RenderElement(result, frames, position);
+                    return RenderElement(context, frames, position);
                 case RenderTreeFrameType.Attribute:
-                    return RenderAttributes(result, frames, position, 1);
+                    throw new InvalidOperationException($"Attributes should only be encountered within {nameof(RenderElement)}");
```

Comment:
I think this is true, and all the tests do still pass. @javiercn, do you have any concerns about this?

Reply:
Do we have a test with more than 1 attribute? I thought we iterated over the attributes that way, but I might be wrong.

Reply:
The new tests for prerendering `<select>` put multiple attributes on a single element and work OK.